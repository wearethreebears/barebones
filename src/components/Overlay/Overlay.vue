<template>
  <div :data-component="$options.name" :class="part['component']">
    <div :class="part['componentInner']">
      <slot />
    </div>
  </div>
</template>

<script lang="ts">
export default { name: 'Overlay' }
</script>

<script setup lang="ts">
import { watch, toRefs } from 'vue';
import useComponent from "@barebones/composables/useComponent";

import { classProps } from "@barebones-local/Overlay/Overlay.classes";
import parts from "@barebones-local/Overlay/Overlay";

const props = defineProps({
  ...classProps,
});

const emit = defineEmits(['visibility:change']);

const { part } = useComponent(parts, props, classProps);

const { state } = toRefs(props)

watch(state, async (newState: string): Promise<void> => {
    // nextTick doesn't wait for visibility transition
    await setTimeout(() => {
      emit('visibility:change', newState)
    }, 100);
})
</script>