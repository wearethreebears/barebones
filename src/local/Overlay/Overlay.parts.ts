import type { Parts } from "@barebones/types/component";
import type { Overlay } from "@barebones/types/components/Overlay";

// Base Classes

export default function OverlayLocal(): Overlay {
  const parts: Parts = {
    component: {
      state: {
        default: ["transition-all duration-300 opacity-0 invisible"],
        open: ["transition-all duration-300 opacity-100 visible"],
      },
      color: {
        default: [
          "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500",
        ],
      },
    },
    overlayInner: {
      color: {
        default: [
          "w-full h-full flex justify-center items-start overflow-y-auto relative",
        ],
      },
    },
  };

  return {
    parts,
  };
}
