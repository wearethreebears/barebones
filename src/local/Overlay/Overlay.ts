import type { Parts } from '@barebones/types/component';

// Base Classes
const ComponentDefault = 'bg-opacity-75 z-50 fixed h-full w-full top-0 left-0';
const ComponentStateDefault = 'transition-all duration-300';
const ComponentInnerDefault= 'w-full h-full flex justify-center items-start overflow-y-auto relative';

const parts: Parts = {
  component: {
    state: {
      default: [ComponentStateDefault, 'opacity-0 invisible'],
      open: [ComponentStateDefault, 'opacity-100 visible']
    },
    color: {
      default: [ComponentDefault, 'bg-slate-500']
    }
  },
  componentInner: {
    color: {
      default: [ComponentInnerDefault]
    }
  },
}

export default parts;
