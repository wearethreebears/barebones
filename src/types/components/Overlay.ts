import type { Parts } from "@barebones/types/component";

export interface Overlay {
  parts: Parts;
}
