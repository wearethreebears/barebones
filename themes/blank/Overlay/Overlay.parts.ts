import type { Parts } from "@barebones/types/component";
import type { Overlay } from "@barebones/types/components/Overlay";

// Base Classes

export default function OverlayLocal(): Overlay {
  const parts: Parts = {
    component: {},
    overlayInner: {},
  };

  return {
    parts,
  };
}
