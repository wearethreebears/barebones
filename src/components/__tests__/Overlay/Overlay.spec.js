import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { allComponentPartClassesToDisplayCorrectly } from "@barebones/components/__tests__/helpers";
import Overlay from "@barebones/components/Overlay/Overlay.vue";

import OverlayLocal from "@barebones-local/Overlay/Overlay.parts";
const { parts } = OverlayLocal();

allComponentPartClassesToDisplayCorrectly(Overlay, parts);

const defaultSlot = "Hello world";

describe("Overlay", () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = mount(Overlay, {
      slots: {
        default: defaultSlot
      }
    });
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the correct slot", () => {
    expect(wrapper.find('[data-part="overlayInner"]').html()).contains(defaultSlot)
  });

});
