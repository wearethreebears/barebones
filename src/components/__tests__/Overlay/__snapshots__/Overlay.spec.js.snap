// Vitest Snapshot v1

exports[`Overlay > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="transition-all duration-300 opacity-0 invisible bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500"
        data-component="BonesOverlay"
        data-part="component"
      >
        <div
          class="w-full h-full flex justify-center items-start overflow-y-auto relative"
          data-overlay-inner=""
          data-part="overlayInner"
        >
          
          Hello world
          
        </div>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": {
          "_ctx": [Circular],
          "default": [Function],
        },
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "color": "default",
            "state": "default",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "emit": [Function],
            "part": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {
                "component": [
                  "transition-all duration-300",
                  "opacity-0 invisible",
                  "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0",
                  "bg-slate-500",
                ],
                "overlayInner": [
                  "w-full h-full flex justify-center items-start overflow-y-auto relative",
                ],
              },
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    ReactiveEffect {
                      "active": true,
                      "deps": [
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "parts": {
              "component": {
                "color": {
                  "default": [
                    "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0",
                    "bg-slate-500",
                  ],
                },
                "state": {
                  "default": [
                    "transition-all duration-300",
                    "opacity-0 invisible",
                  ],
                  "open": [
                    "transition-all duration-300",
                    "opacity-100 visible",
                  ],
                },
              },
              "overlayInner": {
                "color": {
                  "default": [
                    "w-full h-full flex justify-center items-start overflow-y-auto relative",
                  ],
                },
              },
            },
            "props": {
              "color": "default",
              "state": "default",
            },
            "state": ObjectRefImpl {
              "__v_isRef": true,
              "_defaultValue": undefined,
              "_key": "state",
              "_object": {
                "color": "default",
                "state": "default",
              },
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": {
            "visibility:change": null,
          },
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "color": "default",
            "state": "default",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "color": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "state": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
            },
            [
              "state",
              "color",
            ],
          ],
          "provides": {},
          "proxy": {
            "color": "default",
            "state": "default",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {
                    "component": [
                      "transition-all duration-300",
                      "opacity-0 invisible",
                      "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0",
                      "bg-slate-500",
                    ],
                    "overlayInner": [
                      "w-full h-full flex justify-center items-start overflow-y-auto relative",
                    ],
                  },
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    ReactiveEffect {
                      "active": true,
                      "deps": [
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "deps": [
                  Set {
                    [Circular],
                    ReactiveEffect {
                      "active": true,
                      "computed": ComputedRefImpl {
                        "__v_isReadonly": true,
                        "__v_isRef": true,
                        "_cacheable": true,
                        "_dirty": false,
                        "_setter": [Function],
                        "_value": {
                          "component": [
                            "transition-all duration-300",
                            "opacity-0 invisible",
                            "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0",
                            "bg-slate-500",
                          ],
                          "overlayInner": [
                            "w-full h-full flex justify-center items-start overflow-y-auto relative",
                          ],
                        },
                        "dep": Set {
                          ReactiveEffect {
                            "active": true,
                            "allowRecurse": true,
                            "deps": [
                              [Circular],
                            ],
                            "fn": [Function],
                            "onTrack": undefined,
                            "onTrigger": undefined,
                            "parent": undefined,
                            "scheduler": [Function],
                          },
                        },
                        "effect": [Circular],
                      },
                      "deps": [
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {
              "default": [Function],
            },
          },
          "setupState": {
            "emit": [Function],
            "part": {
              "component": [
                "transition-all duration-300",
                "opacity-0 invisible",
                "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0",
                "bg-slate-500",
              ],
              "overlayInner": [
                "w-full h-full flex justify-center items-start overflow-y-auto relative",
              ],
            },
            "parts": {
              "component": {
                "color": {
                  "default": [
                    "bg-opacity-75 z-50 fixed h-full w-full top-0 left-0",
                    "bg-slate-500",
                  ],
                },
                "state": {
                  "default": [
                    "transition-all duration-300",
                    "opacity-0 invisible",
                  ],
                  "open": [
                    "transition-all duration-300",
                    "opacity-100 visible",
                  ],
                },
              },
              "overlayInner": {
                "color": {
                  "default": [
                    "w-full h-full flex justify-center items-start overflow-y-auto relative",
                  ],
                },
              },
            },
            "props": {
              "color": "default",
              "state": "default",
            },
            "state": "default",
          },
          "slots": {
            "default": [Function],
          },
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": ,
                    "appContext": null,
                    "children": [
                      {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "Hello world",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": Hello world,
                        "key": null,
                        "patchFlag": 0,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 8,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": Symbol(Text),
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": ,
                    "key": "_default",
                    "patchFlag": -2,
                    "props": {
                      "key": "_default",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 16,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Fragment),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="w-full h-full flex justify-center items-start overflow-y-auto relative"
                  data-overlay-inner=""
                  data-part="overlayInner"
                >
                  
                  Hello world
                  
                </div>,
                "key": null,
                "patchFlag": 2,
                "props": {
                  "class": "w-full h-full flex justify-center items-start overflow-y-auto relative",
                  "data-overlay-inner": "",
                  "data-part": "overlayInner",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Hello world",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Hello world,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": ,
                    "appContext": null,
                    "children": [
                      {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "Hello world",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": Hello world,
                        "key": null,
                        "patchFlag": 0,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 8,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": Symbol(Text),
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": ,
                    "key": "_default",
                    "patchFlag": -2,
                    "props": {
                      "key": "_default",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 16,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Fragment),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="w-full h-full flex justify-center items-start overflow-y-auto relative"
                  data-overlay-inner=""
                  data-part="overlayInner"
                >
                  
                  Hello world
                  
                </div>,
                "key": null,
                "patchFlag": 2,
                "props": {
                  "class": "w-full h-full flex justify-center items-start overflow-y-auto relative",
                  "data-overlay-inner": "",
                  "data-part": "overlayInner",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "dynamicProps": [
              "data-component",
            ],
            "el": <div
              class="transition-all duration-300 opacity-0 invisible bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500"
              data-component="BonesOverlay"
              data-part="component"
            >
              <div
                class="w-full h-full flex justify-center items-start overflow-y-auto relative"
                data-overlay-inner=""
                data-part="overlayInner"
              >
                
                Hello world
                
              </div>
            </div>,
            "key": null,
            "patchFlag": 10,
            "props": {
              "class": "transition-all duration-300 opacity-0 invisible bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500",
              "data-component": "BonesOverlay",
              "data-part": "component",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/Overlay/Overlay.vue",
            "__hmrId": "19b7921e",
            "components": {},
            "emits": [
              "visibility:change",
            ],
            "name": "BonesOverlay",
            "props": {
              "color": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "state": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 9,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="transition-all duration-300 opacity-0 invisible bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500"
          data-component="BonesOverlay"
          data-part="component"
        >
          <div
            class="w-full h-full flex justify-center items-start overflow-y-auto relative"
            data-overlay-inner=""
            data-part="overlayInner"
          >
            
            Hello world
            
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 36,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/Overlay/Overlay.vue",
          "__hmrId": "19b7921e",
          "components": {},
          "emits": [
            "visibility:change",
          ],
          "name": "BonesOverlay",
          "props": {
            "color": {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
            "state": {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 8,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="transition-all duration-300 opacity-0 invisible bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500"
          data-component="BonesOverlay"
          data-part="component"
        >
          <div
            class="w-full h-full flex justify-center items-start overflow-y-auto relative"
            data-overlay-inner=""
            data-part="overlayInner"
          >
            
            Hello world
            
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 4,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": {
    "color": "default",
    "state": "default",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    class="transition-all duration-300 opacity-0 invisible bg-opacity-75 z-50 fixed h-full w-full top-0 left-0 bg-slate-500"
    data-component="BonesOverlay"
    data-part="component"
  >
    <div
      class="w-full h-full flex justify-center items-start overflow-y-auto relative"
      data-overlay-inner=""
      data-part="overlayInner"
    >
      
      Hello world
      
    </div>
  </div>,
}
`;
